api/openapi_server/controllers/auth_controller.py

```diff
@@ -58,6 +58,22 @@ def get_token_auth_header():
     # return token
     token = parts[1]
     return token
+def delete_user(email: str):
+    with DataAccessLayer.session() as session:
+        user = session.query(User).filter_by(email=email).first()
+        if user is None:
+            raise AuthError({
+                "message": "User with this email does not exist."
+            }, 404)
+
+        try:
+            session.delete(user)
+            session.commit()
+        except IntegrityError:
+            session.rollback()
+            raise AuthError({
+                "message": "Failed to delete user"
+            }, 500)
```

Comment:
I added a very similar method that removes users from the db and AWS Cognito in #641, `admin_controller.removeUser`. You could almost use it here, but you'd have to handle the `AuthError` that is thrown when the user is not found in the AWS Cognito pool. 

This is optional, but it would reduce duplicate logic if you modify `removeUser` to avoid re-raising the `UserNotFoundException` `AuthError` and reusing the logic here. (also I should have named the function `remove_user` 😅).

Reply:
Great thanks for providing some insights on how to accomplish this 😄